mlflow/genai/agent_server/__init__.py

```diff
@@ -0,0 +1,455 @@
+                    "request_size": len(json.dumps(data)),
+                    "stream_requested": data.get("stream", False),
+                },
+            )
+
+            is_streaming = data.get("stream", False)
+            return_trace = data.get("databricks_options", {}).get("return_trace", False)
+
+            request_data = {k: v for k, v in data.items() if k != "stream"}
+
+            try:
+                request_data = self.validator.validate_and_convert_request(request_data)
+            except ValueError as e:
+                raise HTTPException(
+                    status_code=400,
+                    detail=f"Invalid parameters for {self.agent_type}: {e}",
+                )
+
+            if is_streaming:
+                return await self._handle_stream_request(request_data, start_time, return_trace)
+            else:
+                return await self._handle_invoke_request(request_data, start_time, return_trace)
+
+        @self.app.get("/health")
+        async def health_check() -> dict[str, str]:
+            """Health check endpoint for frontend connection testing"""
+            return {"status": "healthy"}
+
+    async def _handle_invoke_request(
+        self, data: dict[str, Any], start_time: float, return_trace: bool
+    ) -> dict[str, Any]:
+        """Handle non-streaming invoke requests"""
+        if _invoke_function is None:
+            raise HTTPException(status_code=500, detail="No invoke function registered")
+
+        func = _invoke_function
+        func_name = func.__name__
+
+        try:
+            with mlflow.start_span(name=f"{func_name}") as span:
+                span.set_inputs(data)
+                if inspect.iscoroutinefunction(func):
+                    result = await func(data)
+                else:
+                    result = func(data)
+
+                result = self.validator.validate_and_convert_result(result)
+                duration_ms = round((time.time() - start_time) * 1000, 2)
+                span.set_attribute("duration_ms", duration_ms)
+                if self.agent_type == "agent/v1/responses":
+                    span.set_attribute("mlflow.message.format", "openai")
+                span.set_outputs(result)
+
+                if return_trace:
+                    databricks_output = self._get_databricks_output(span.trace_id)
+                    result["databricks_output"] = databricks_output
+
+            self.logger.info(
+                "Response sent",
+                extra={
+                    "endpoint": "invoke",
+                    "duration_ms": duration_ms,
+                    "response_size": len(json.dumps(result)),
+                    "function_name": func_name,
+                    "return_trace": return_trace,
+                },
+            )
+
+            return result
+
+        except Exception as e:
+            duration_ms = round((time.time() - start_time) * 1000, 2)
+            span.set_attribute("duration_ms", duration_ms)
```

Comment:
This is not within the `with mlflow.start_span` context, it's possible the span doesn't exist